### ISSUE TITLE
[Question]: How do you use a finetuned model for mobile

### ISSUE THREAD

**Issue by shaon-chowdhury** [microsoft/Olive#1186]
I've created a fine-tuned model using the examples (target had to be CUDA) and would would now like to use it for mobile. How do I go about this?

**Comment by jambayk** [comment]
Could you give us some background on what model and what fine tuning method you used? 

We have an example for phi3 https://github.com/microsoft/Olive/tree/main/examples/phi3. Does that help?

**Comment by samuel100** [comment]
@shaon-chowdhury [This notebook](https://github.com/microsoft/Olive/blob/main/examples/getting_started/olive-awq-ft-llama.ipynb) shows how to:

1. **Quantize** Llama-3.2 using the AWQ Algorithm
2. **Fine-tune** the quantized model - [fine-tuning *after* quantization gives better model quality as some of the loss from the quantization process can be recovered](https://onnxruntime.ai/blogs/olive-quant-ft). 
3. Run Olive's **automatic optimizer** on the fine-tuned model and adapter to capture the ONNX graph and optimize the graph.
4. Inference the model and adapter on the ONNX Runtime using the Generate API. You'll see in the example that the ONNX Runtime allows you to do MultiLoRA serving.  For more details, read [this announcement](https://onnxruntime.ai/blogs/multilora). 

>[!NOTE]
>Whilst the example uses Llama-3.2, you can choose a different model.

The model is optimized for CPU, which will run on mobile using ORT. We'd need more details on the hardware (e.g. is an NPU available?) to see if you can optimize further. Please open as a discussion.
